homeassistant/components/ovo_energy/__init__.py

```diff
@@ -0,0 +1,112 @@
+"""Support for OVO Energy."""
+import logging
+from typing import Any, Dict
+
+import aiohttp
+from ovoenergy.ovoenergy import OVOEnergy
+
+from homeassistant.components.ovo_energy.const import (
+    CONF_ACCOUNT_ID,
+    DATA_OVO_ENERGY_CLIENT,
+    DOMAIN,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistantType, config: ConfigType) -> bool:
+    """Set up the OVO Energy components."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
+    """Set up OVO Energy from a config entry."""
+
+    client = OVOEnergy()
+
+    try:
+        await client.authenticate(entry.data[CONF_USERNAME], entry.data[CONF_PASSWORD])
+    except aiohttp.ClientError as exception:
+        _LOGGER.warning(exception)
+        raise ConfigEntryNotReady from exception
+
+    instance_key = f"{DOMAIN}_{entry.data[CONF_ACCOUNT_ID]}"
+    hass.data.setdefault(instance_key, {})[DATA_OVO_ENERGY_CLIENT] = client
```

Comment:
It is better to namespace by domain and use the config entry as the unique ID to differentiate between the different config entries.

```suggestion
    hass.data.setdefault(DOMAIN, {})
    hass.data[DOMAIN][entry.entry_id] = client
```

Of course, this needs adjusting across the rest of the integrations.